homeassistant/components/habitica/todo.py

```diff
@@ -117,20 +117,24 @@ async def async_move_todo_item(
         """Move an item in the To-do list."""
         if TYPE_CHECKING:
             assert self.todo_items
+        tasks_order = (
+            self.coordinator.data.user.tasksOrder.todos
+            if self.entity_description.key is HabiticaTodoList.TODOS
+            else self.coordinator.data.user.tasksOrder.dailys
+        )
 
         if previous_uid:
-            pos = self.todo_items.index(
-                next(item for item in self.todo_items if item.uid == previous_uid)
-            )
-            if pos < self.todo_items.index(
-                next(item for item in self.todo_items if item.uid == uid)
-            ):
+            pos = tasks_order.index(UUID(previous_uid))
+            if pos < tasks_order.index(UUID(uid)):
                 pos += 1
+
         else:
             pos = 0
 
         try:
-            await self.coordinator.habitica.reorder_task(UUID(uid), pos)
+            tasks_order = (
```

Comment:
```suggestion
            tasks_order[:] = (
```
otherwise it is only assigning the new list to the local `tasks_order`.